tests/test_aircon.py

```diff
@@ -1,228 +1,144 @@
-    backend_selector_mock: BackendSelectorMock,
-    auth_mock: MagicMock,
-):
-    mock_appliance_http_get(
-        appliance_http_client_mock, backend_selector_mock, SAID, DATA1
-    )
-    mock_appliance_http_post(appliance_http_client_mock, backend_selector_mock)
-    appliance_http_client_mock.create_session(asyncio.get_event_loop())
-    CONNECT_HTTP_CALLS = 2
-    aircon = Aircon(
-        backend_selector_mock,
-        auth_mock,
-        SAID,
-        appliance_http_client_mock.session,
-    )
-    await aircon.connect()
 
-    await aircon.set_power_on(True)
-    assert_appliance_setter_call(
-        appliance_http_client_mock,
-        SAID,
-        {"Sys_OpSetPowerOn": "1"},
-        CONNECT_HTTP_CALLS + 1,
-    )
 
-    await aircon.set_power_on(False)
-    assert_appliance_setter_call(
-        appliance_http_client_mock,
-        SAID,
-        {"Sys_OpSetPowerOn": "0"},
-        CONNECT_HTTP_CALLS + 2,
-    )
+CURR_DIR = Path(__file__).parent
+DATA_DIR = CURR_DIR / "data"
 
-    await aircon.set_temp(30)
-    assert_appliance_setter_call(
-        appliance_http_client_mock,
-        SAID,
-        {"Sys_OpSetTargetTemp": "300"},
-        CONNECT_HTTP_CALLS + 3,
-    )
+aircon_data_file = DATA_DIR / "aircon_data.json"
+aircon_data = json.loads(aircon_data_file.read_text())
 
-    await aircon.set_humidity(45)
-    assert_appliance_setter_call(
-        appliance_http_client_mock,
-        SAID,
-        {"Sys_OpSetTargetHumidity": "45"},
-        CONNECT_HTTP_CALLS + 4,
-    )
+DATA1 = aircon_data["DATA1"]
+DATA2 = aircon_data["DATA2"]
 
-    await aircon.set_mode(Mode.Cool)
-    assert_appliance_setter_call(
-        appliance_http_client_mock,
-        SAID,
-        {"Cavity_OpSetMode": "1"},
-        CONNECT_HTTP_CALLS + 5,
-    )
 
-    await aircon.set_fanspeed(FanSpeed.Auto)
-    assert_appliance_setter_call(
-        appliance_http_client_mock,
-        SAID,
-        {"Cavity_OpSetFanSpeed": "1"},
-        CONNECT_HTTP_CALLS + 6,
-    )
-
-    await aircon.set_h_louver_swing(True)
-    assert_appliance_setter_call(
-        appliance_http_client_mock,
-        SAID,
-        {"Cavity_OpSetHorzLouverSwing": "1"},
-        CONNECT_HTTP_CALLS + 7,
-    )
-
-    await aircon.set_turbo_mode(False)
-    assert_appliance_setter_call(
-        appliance_http_client_mock,
-        SAID,
-        {"Cavity_OpSetTurboMode": "0"},
-        CONNECT_HTTP_CALLS + 8,
-    )
-
-    await aircon.set_eco_mode(False)
-    assert_appliance_setter_call(
-        appliance_http_client_mock,
-        SAID,
-        {"Sys_OpSetEcoModeEnabled": "0"},
-        CONNECT_HTTP_CALLS + 9,
-    )
-
-    await aircon.set_quiet_mode(False)
-    assert_appliance_setter_call(
-        appliance_http_client_mock,
-        SAID,
-        {"Sys_OpSetQuietModeEnabled": "0"},
-        CONNECT_HTTP_CALLS + 10,
-    )
-
-    await aircon.set_display_on(True)
-    assert_appliance_setter_call(
-        appliance_http_client_mock,
+async def test_attributes():
```

Comment:
```suggestion
@aioresponses()
async def test_request(mocked_aioresponses):
```
I think having this as a fixture instead of the context manager makes the tests a bit cleaner, and also reduce nesting